### ISSUE TITLE
Notes: GRAD Cyrillic edits

### ISSUE THREAD

**Issue by alexeiva** [googlefonts/googlesans#449]
I am working on this branch https://github.com/googlefonts/googlesans/tree/alexeiva-cyr-GRAD-fix
to cleanup various interpolation errors in Cyrillic GRAD masters. 

The list of glyphs I amended is included in the commit messages. 

Here are some irregularities I found additionally to my review between Latin and Cyrillic stems in `opsz18-wght380-GRAD200`:

<img width="1980" alt="Screenshot 2022-01-21 at 22 16 20" src="https://user-images.githubusercontent.com/16634089/150586935-d94fb392-c85e-46d6-b542-aa0ad61519a6.png">

<img width="1965" alt="Screenshot 2022-01-21 at 22 18 49" src="https://user-images.githubusercontent.com/16634089/150587259-5b483698-9296-42a3-9033-f2a9e6a7b432.png">

There is a 3 unit difference in UC and 4 units in LC, that can be resolved using auto-transformation

**Comment by chrissimpkins** [comment]
> There is a 3 unit difference in UC and 4 units in LC, that can be resolved using auto-transformation

Thanks Alexei!  Is this a question of whether to do so?  Or have you already done it?

**Comment by alexeiva** [comment]
Here are all inconsistencies I discovered in Cyrillic stems. I am just noting them down to batch-fix later. 

**fixed:**
```
lowercase, whole CYR set:
G-50 17 and 18
64 to 65 verticals 
57 to 59 horizontals
```
Inconsistencies that I have resolved:
```
Small Caps, single glyph inconsistencies
G-50 17 and 18
n.sc - incorrect crossbar 58 not 64
t.sc incorrect stem 65 not 63
de.sc - incorrect horizontal bar

G200 18 and 17
n.sc 
108 not 102

G200 18 and 17 
pe-cy 108 not 102
sha-cy 108 not 92
Sha-cy 117 not 116

Uppercase, whole CYR set
G200 17
125 to 122 verticals
121 to 116 horizontals

G200 17  lowercase
114 to 118 verticals 

G200 18 lowercase
115 to 119 verticals
```

**Comment by alexeiva** [comment]
My latest commit caused an error in an armenian lookup. I edited the fonts in prepolator.
Not sure how this happened, will revert the commit. 

https://github.com/googlefonts/googlesans/runs/5027376804?check_suite_focus=true

```
FONTTOOLS_GPOS_COMPACT_MODE=5 fontmake --verbose WARNING -m GoogleSans/GoogleSans-Italic.designspace -o variable --output-dir ../build/GoogleSans/variable
fontmake: Error: In 'GoogleSans/GoogleSans.designspace': Generating fonts from Designspace failed: 

Couldn't merge the fonts, because some values were different, but should have
been the same. This happened while performing the following operation:
GSUB.table.LookupList.Lookup[9][0].BacktrackCoverage[0].glyphs[0]

The problem is likely to be in Google Sans Text Regular GRAD-50 Regular:
Expected to see [0]==ech-arm.sc, instead saw ech_vew-arm.liga.sc
```

**Comment by madig** [comment]
Hot take: check the public.glyphOrder lib keys of all your UFOs listed in GoogleSans/GoogleSans.designspace, they must be identical within each family.

**Comment by alexeiva** [comment]
@madig Do you mean these keys?
<img width="1666" alt="Screenshot 2022-02-03 at 11 43 19" src="https://user-images.githubusercontent.com/16634089/152309333-07196d40-2c0f-4654-9191-0802fe5e4732.png">

**Comment by alexeiva** [comment]
would it be okay if  convert the UFOs to glyphs via `scripts/gs-ufo2glyphs.py`, 
and provide a glyphs file instead?

**Comment by madig** [comment]
Yeah, the one on the left. Yes that's fine :)

**Comment by alexeiva** [comment]
I have [reverted](https://github.com/googlefonts/googlesans/commit/8a58edc3efef0300eb4cf7b0709f4d773d17c051) the original `lib.plist`, and the opsz17 GRAD=50 file now builds. 

Will do the same for other styles.

This is what caused the issue:
 I added a smart CYR glyph list in Robofont, that changed `public.glyphOrder`
<img width="765" alt="Screenshot 2022-02-03 at 12 06 42" src="https://user-images.githubusercontent.com/16634089/152312668-af09f2c0-2598-461e-9df0-f11d51fdb2d6.png">

**Comment by madig** [comment]
That would work as well. That, or running `python3 scripts/gs-normalize-designspace.py` inside a venv. If all you're doing is modifying existing glyphs geometrically (not adding or removing anything), there should be nothing lib.plist that should change.

**Comment by alexeiva** [comment]
My Roman .glyphs file is complete, Italics are in the works. I am testing the merging process. 

Completed Step 1, and there is a `maximum recursion depth` error while running step 2 from  [Maintainer.md](https://github.com/googlefonts/googlesans/blob/main/docs/MAINTAINER.md#source-conventions)

```
 ~/github.com/googlefonts/googlesans > python scripts/gs-glyphs2ufo.py source/GoogleSans/staging/*.glyphs
Traceback (most recent call last):
  File "/Users/alexeiva/github.com/googlefonts/googlesans/scripts/gs-glyphs2ufo.py", line 76, in <module>
    normalize.scrub_designspace(designspace, ROOT_DIR)
  File "/Users/alexeiva/github.com/googlefonts/googlesans/scripts/internal/normalize.py", line 40, in scrub_designspace
    ot_categories = infer_opentype_categories(designspace.default.font)
  File "/Users/alexeiva/github.com/googlefonts/googlesans/scripts/internal/normalize.py", line 301, in infer_opentype_categories
    pf(font=ufo_copy)
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 12, in __call__
    if super().__call__(font, glyphSet):
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/base.py", line 195, in __call__
    if include(glyph) and filter_(glyph):
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 24, in filter
    flattened_tuples = _flattenComponent(self.context.glyphSet, comp)
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 46, in _flattenComponent
    flattened_components = _flattenComponent(glyphSet, nested)
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 46, in _flattenComponent
    flattened_components = _flattenComponent(glyphSet, nested)
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 46, in _flattenComponent
    flattened_components = _flattenComponent(glyphSet, nested)
  [Previous line repeated 987 more times]
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufo2ft/filters/flattenComponents.py", line 40, in _flattenComponent
    if not glyph.components or len(glyph) > 0:
  File "/Users/alexeiva/virtualenvs/gftools/lib/python3.9/site-packages/ufoLib2/objects/glyph.py", line 86, in __len__
    return len(self.contours)
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by madig** [comment]
You most likely have a self-referencing component somewhere. Try killing all background layers.

**Comment by alexeiva** [comment]
>You most likely have a self-referencing component somewhere. Try killing all background layers.

Tried removing BG layers in the upstream [branch ](https://github.com/googlefonts/googlesans/tree/Berlow-Greek-GRAD-Fixes)
Also tried removing the bracket layer from `y` (it is incompatible in Glyphs App), didn't help. 
<img width="491" alt="Screenshot 2022-02-13 at 12 36 49" src="https://user-images.githubusercontent.com/16634089/153757407-a57df95b-54e9-40e3-bec3-d9bd573d3d03.png">

Could it be `schwadieresis-cy` that is a component of `schwa-cy`, which is a component of `schwa`?

**Comment by alexeiva** [comment]
Found a design error in dje-cy.sc, Dje-cy italics. Will fix this. 

<img width="426" alt="Screenshot 2022-02-13 at 18 20 51" src="https://user-images.githubusercontent.com/16634089/153760144-cd71b053-56c4-471b-a74e-efdf002c894e.png">

**Comment by alexeiva** [comment]
Italic stems in small caps are narrower. Is this intentional?

<img width="588" alt="Screenshot 2022-02-13 at 14 31 05" src="https://user-images.githubusercontent.com/16634089/153832788-e6cc95d3-9d7c-43b2-812d-b3cdf95ba6fd.png">

<img width="611" alt="Screenshot 2022-02-13 at 14 31 12" src="https://user-images.githubusercontent.com/16634089/153832821-ee46eaf7-338a-43e5-b255-359282d7376c.png">

**Comment by chrissimpkins** [comment]
> Italic stems in small caps are narrower. Is this intentional?
> 
> 
> 
> <img width="588" alt="Screenshot 2022-02-13 at 14 31 05" src="https://user-images.githubusercontent.com/16634089/153832788-e6cc95d3-9d7c-43b2-812d-b3cdf95ba6fd.png">
> 
> 
> 
> <img width="611" alt="Screenshot 2022-02-13 at 14 31 12" src="https://user-images.githubusercontent.com/16634089/153832821-ee46eaf7-338a-43e5-b255-359282d7376c.png">
> 
> 
> 
> 
> 
> 

@dberlow can you answer this question for Alexei?

**Comment by alexeiva** [comment]
Different point endings in Italic. Which is correct?
Glyph `K`:
<img width="495" alt="Screenshot 2022-02-15 at 14 44 31" src="https://user-images.githubusercontent.com/16634089/154055801-c0aef02b-f3e5-42e5-ac39-df5a09baa37b.png">

Glyph `K.alt`. I would prefer this for italic — better aesthetically and less trouble with interpolation. 
<img width="585" alt="Screenshot 2022-02-15 at 14 44 07" src="https://user-images.githubusercontent.com/16634089/154055822-a696fbc7-8cf1-4ac9-8a04-807dc2bfeb4a.png">

**Comment by chrissimpkins** [comment]
@dberlow will you please weigh in on Alexei's questions about outlines in this thread?

**Comment by alexeiva** [comment]
Italic
cyrillic breve's in 18opsz should be shifter
<img width="802" alt="Screenshot 2022-02-16 at 17 57 23" src="https://user-images.githubusercontent.com/16634089/154291834-0075760a-f777-4d99-8683-7494b01ae4ee.png">


Upright — ok
<img width="921" alt="Screenshot 2022-02-16 at 17 57 27" src="https://user-images.githubusercontent.com/16634089/154291551-dc30bd32-bb3d-44d2-9fba-aeaaa06919ec.png">

**Comment by alexeiva** [comment]
There is also a stems issue in GRAD200 Italic.


<img width="607" alt="Screenshot 2022-02-17 at 10 18 00" src="https://user-images.githubusercontent.com/16634089/154425422-1d983be3-639c-48de-a437-a42123ed94dc.png">
<img width="566" alt="Screenshot 2022-02-17 at 10 17 57" src="https://user-images.githubusercontent.com/16634089/154425432-26e25207-73b8-45c1-8c5c-95dbbc8651d2.png">


<img width="765" alt="Screenshot 2022-02-17 at 10 23 11" src="https://user-images.githubusercontent.com/16634089/154425638-57fff1e3-a98a-4d17-8834-f4469f148ab7.png">
<img width="541" alt="Screenshot 2022-02-17 at 10 23 08" src="https://user-images.githubusercontent.com/16634089/154425644-c3c604b5-e503-462d-9f7a-564550458316.png">

**Comment by chrissimpkins** [comment]
Let's chat about these issues during the meeting tomorrow since we can't get a response to your comments and questions Alexei.

**Comment by alexeiva** [comment]
Let's also discuss the o, c dimensions, since they affect Cyrillic. 
I'm not sure about this compensation, I would keep the stems the same size between Regular and Italic. 

<img width="647" alt="Screenshot 2022-02-17 at 22 31 17" src="https://user-images.githubusercontent.com/16634089/154556720-709e9b42-0015-4c03-a858-8fba8032c04b.png">


<img width="718" alt="Screenshot 2022-02-17 at 22 31 27" src="https://user-images.githubusercontent.com/16634089/154556706-fb87680d-9196-4445-b686-8e7bb65c8270.png">

**Comment by alexeiva** [comment]
Bold vs Italic
<img width="769" alt="Screenshot 2022-02-17 at 22 34 22" src="https://user-images.githubusercontent.com/16634089/154558240-29ea216d-2730-4a03-bee0-b8adf52e837f.png">

GRAD 200 and GRAD -50

<img width="764" alt="Screenshot 2022-02-17 at 22 40 46" src="https://user-images.githubusercontent.com/16634089/154558156-1c292fdb-5e98-4ecb-973a-affabf3d89f4.png">
<img width="794" alt="Screenshot 2022-02-17 at 22 39 40" src="https://user-images.githubusercontent.com/16634089/154558162-63d1d6f1-32c8-49c2-ba83-b644af96084b.png">
<img width="850" alt="Screenshot 2022-02-17 at 22 37 28" src="https://user-images.githubusercontent.com/16634089/154558166-3d9c6d6a-5b53-4c01-ae9d-caa01b8b015e.png">

**Comment by dberlow** [comment]
Hey guys,

Sorry not to respond. I’ve been investigating these things as they come up and waiting for today’s meeting to address them verbally, but was then told Chris is in vacation, so I am coming back to write it here.

Italic companion styles are almost always optically corrected versus the regular, in ways that were evident in the original styles Font Bureau began with. I can confirm the original styles were drawn with lighter italics than the regular. Thus the grades will follow, with lighter italic grades than regular.

What situations exist between small caps should be consistent with the other glyphs, but their grading was a big question for a few months and it was left to me to determine.

If there are any errors in the work, please let us know, with your review knowing the default fonts were confirmed in their design before we started, and we did not change them.

On Feb 14, 2022, at 8:49 AM, Chris Simpkins ***@***.***> wrote:

﻿
Italic stems in small caps are narrower. Is this intentional?

 
@dberlow can you answer this question for Alexei?

**Comment by alexeiva** [comment]
I'm not sure how to deal with the Italic `o.sc` in Grad -50, it has reverse contrast, that isn't supported in other small caps.
<img width="562" alt="Screenshot 2022-02-18 at 08 42 35" src="https://user-images.githubusercontent.com/16634089/154625004-0d782246-9ceb-49ca-ac6a-4ef2f5e2dde9.png">
<img width="575" alt="Screenshot 2022-02-18 at 08 42 30" src="https://user-images.githubusercontent.com/16634089/154625062-b3cac3b2-c3e8-4d38-9489-708ac2cb6aca.png">

I am basing the Cyrillic stem values on `omegatonos.sc`
<img width="895" alt="Screenshot 2022-02-18 at 08 57 04" src="https://user-images.githubusercontent.com/16634089/154626373-51af27ba-c845-44d3-bc89-b40121146f87.png">

**Comment by dberlow** [comment]
Thanks! o.sc and q.sc revised and committed.
Please close issue if satisfied.

**Comment by alexeiva** [comment]
@dberlow Can you point me to the updated commit please?

Also, I need to understand the basic stem dimensions in Text Italic GRAD 200 to coordinate with Cyrillic. 

<img width="588" alt="153832788-e6cc95d3-9d7c-43b2-812d-b3cdf95ba6fd" src="https://user-images.githubusercontent.com/16634089/154689440-cdc1c4a6-575d-488b-b4b1-fca0b7e8bd96.png">

<img width="611" alt="153832821-ee46eaf7-338a-43e5-b255-359282d7376c" src="https://user-images.githubusercontent.com/16634089/154689430-888b213b-7420-4f3d-a5a9-02157d836372.png">

**Comment by dberlow** [comment]
Three days ago it cloned this repo, today's changes were committed to the main branch. I tried repeating that just now and the commit or one to any branch was refused. 
[GoogleSansItalic-opsz18-wght380-GRAD-50.ufo.zip](https://github.com/googlefonts/googlesans/files/8098179/GoogleSansItalic-opsz18-wght380-GRAD-50.ufo.zip)
A compressed copy of the edited source is included here.

**Comment by alexeiva** [comment]
@dberlow I switched to .glyphs sources, I've completed my work, and posted the zip here: https://github.com/googlefonts/googlesans/issues/456

I have one more remaining issue — GRAD200 (opsz17 and opsz18) have stem dimension inconsistencies with Roman styles. I left them as they are:



1
<img width="1229" alt="1" src="https://user-images.githubusercontent.com/16634089/154728381-b1c42d4e-f92c-44a2-a9bf-6b3201740cef.png">


2 
<img width="1183" alt="2" src="https://user-images.githubusercontent.com/16634089/154728394-b313b6d5-ee9a-4d82-9025-c3a6eea0db0a.png">

**Comment by chrissimpkins** [comment]
@dberlow David, I spoke with Alexei today. He identified GRAD +200 master stem width and terminal ending inconsistencies in the Latin and Greek during his work and changed the Latin `C`, `c` forms as part of his edits (due to use as components in Cyrillic).  He will open up a new issue to document this finding.  I'd like to discuss/review this with you before we merge the Latin changes that he made. Let's also discuss if any of these changes are included in your Greek GRAD -50 / +200 revisions and how we should approach the Latin stem widths in GRAD +200.  Let's chat about this when we get together later this week.  Thanks!
